src/ai/backend/storage/kmanila/__init__.py

```diff
@@ -0,0 +1,272 @@
+import asyncio
+import logging
+from pathlib import Path
+from typing import Any, Final, TypedDict
+
+import aiohttp
+import trafaret as t
+from yarl import URL
+
+from ai.backend.common.etcd import AsyncEtcd
+from ai.backend.common.logging import BraceStyleAdapter
+from ai.backend.common.types import QuotaScopeID
+
+from ..abc import AbstractQuotaModel
+from ..exception import ExecutionError, ExternalError
+from ..types import QuotaConfig
+from ..vfs import BaseQuotaModel, BaseVolume
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))  # type: ignore[name-defined]
+
+
+DEFAULT_MAX_POLL_COUNT: Final = 20
+DEFAULT_VOLUME_SIZE: Final = 1000
+QUOTA_VOLUME_ID_MAP_KEY: Final = "kmanila/volumes"
+
+
+kmanila_request_params = t.Dict(
+    {
+        t.Key("volume_name"): t.String(),
+    }
+).allow_extra("*")
+
+
+class QuotaVolumeMap(TypedDict):
+    volume_id: str
+    volume_name: str
+
+
+class KManilaQuotaModel(BaseQuotaModel):
+    def __init__(
+        self,
+        mount_path: Path,
+        etcd: AsyncEtcd,
+        api_base_url: URL,
+        agent_addrs: list[str],
+        *,
+        access_to: str,
+        access_level: str,
+        kmanila_requestor_id: str,
+        kmanila_requestor_pwd: str,
+        default_project_id: str | None,
+        default_netword_id: str | None,
+        max_poll_count: int,
+    ) -> None:
+        super().__init__(mount_path)
+        self.etcd = etcd
+        self.api_base_url = api_base_url
+        self.agent_addrs = agent_addrs
+        self.access_to = access_to
+        self.access_level = access_level
+        self.kmanila_requestor_id = kmanila_requestor_id
+        self.kmanila_requestor_pwd = kmanila_requestor_pwd
+        self.default_project_id = default_project_id
+        self.default_netword_id = default_netword_id
+        self.max_poll_count = max_poll_count
+
+    async def _get_auth_info(self, user_id: QuotaScopeID) -> dict[str, Any]:
+        async with aiohttp.ClientSession() as sess:
+            rqst_id = self.kmanila_requestor_id
+            rqst_pwd = self.kmanila_requestor_pwd
+            request_body = {
+                "auth": {
+                    "identity": {
+                        "methods": ["password"],
+                        "password": {
+                            "user": {
+                                "domain": {"id": "default"},
+                                "name": rqst_id,
+                                "password": rqst_pwd,
+                            }
+                        },
+                    },
+                    "scope": {
+                        "project": {
+                            "domain": {"id": "default"},
+                            "name": rqst_id,
+                        }
+                    },
+                }
+            }
+            headers = {"Content-Type": "application/json", "Accept": "application/json"}
```

Comment:
when passing request body with `json` parameter aiohttp automatically sets `Content-Type` header to `application/json`.
```suggestion
            headers = {"Accept": "application/json"}
```